### ISSUE TITLE
TypeError while doing tutorial: 'NoneType' object is not callable

### ISSUE THREAD

**Issue by jay-tyler** [secdev/scapy#352]
I'm in an interactive scapy shell (via `sudo scapy`), and am trying to familiarize myself with Scapy via the 'interactive tutorial' in the docs. I'm running into trouble manipulating the packet at the bottom of the [Stacking Layers section](http://scapy.readthedocs.io/en/latest/usage.html#stacking-layers). Calling str() or hexdump() on the packet results in a TypeError.

```
>>> a=Ether()/IP(dst="www.slashdot.org")/TCP()/"GET /index.html HTTP/1.0 \n\n"
>>> str(a)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/packet.py", line 277, in __str__
    return self.build()
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/packet.py", line 354, in build
    p = self.do_build()
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/packet.py", line 341, in do_build
    pkt = self.self_build()
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/packet.py", line 332, in self_build
    p = f.addfield(self, p, val)
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/fields.py", line 72, in addfield
    return s+struct.pack(self.fmt, self.i2m(pkt,val))
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/layers/l2.py", line 105, in i2m
    return MACField.i2m(self, pkt, self.i2h(pkt, x))
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/layers/l2.py", line 97, in i2h
    x = conf.neighbor.resolve(pkt,pkt.payload)
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/layers/l2.py", line 45, in resolve
    return self.resolvers[k](l2inst,l3inst)
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/layers/inet.py", line 782, in inet_register_l3
    return getmacbyip(l3.dst)
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/layers/l2.py", line 80, in getmacbyip
    nofilter=1)
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/sendrecv.py", line 395, in srp1
    a,b=srp(*args,**kargs)
  File "/Users/jtyler/virtualenvs/scapy-playground/lib/python2.7/site-packages/scapy/sendrecv.py", line 377, in srp
    s = conf.L2socket(iface=iface, filter=filter, nofilter=nofilter, type=type)
TypeError: 'NoneType' object is not callable
```
I'm using a MacBook Pro, if that matters.

**Comment by guedou** [comment]
I can't reproduce the issue on the current version. Could you upgrade to a newer version ? If the problem still occur could you reopen the issue ?

**Comment by jay-tyler** [comment]
The issue above is with the current version installed via git. e.g.
```
SEA-ML-JTYLER:scapy-playground jtyler$ pip freeze | grep scapy
You are using pip version 7.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
scapy==2.3.3.dev22
```
Do I have to open another issue to 'reopen' the issue? I don't think I have the appropriate permissions to remove the "Closed" state.

**Comment by guedou** [comment]
Dis you install the pcap and dnet Python modules as explained here http://scapy.readthedocs.io/en/latest/installation.html#platform-specific-instructions ?

**Comment by jay-tyler** [comment]
Ok, hadn't noticed the machine-specific instructions. I just did the brew install, but without 'sudo'. I get a pretty stern warning otherwise:
```
$ sudo brew install --with-python libdnet
Error: Cowardly refusing to 'sudo brew install'
You can use brew with sudo, but only if the brew executable is owned by root.
However, this is both not recommended and completely unsupported so do so at
your own risk.
```
brew installs for pcap and dnet seemed go ok (no console warning messages), but I'm still getting the traceback above. I tried reinstalling all three outside of a virtualenv with the same result.

There's a [somewhat similar unresolved problem](https://github.com/phaethon/scapy/issues/73) for the Py3.x port of this project.
